mlir/include/mlir/Dialect/MIOpen/Pipeline.h

```diff
@@ -0,0 +1,65 @@
+//===- Pipeline.h - MIOpen pipeline creation ---------------===//
+//
+// Part of the MLIR Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file declares the miopen::addPipeline routine to consolidate the
+// MIOpen pipeline creation.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef MLIR_DIALECT_MIOPEN_PIPELINE_H
+#define MLIR_DIALECT_MIOPEN_PIPELINE_H
+
+#include <mlir/Pass/PassManager.h>
+
+#include <string>
+
+namespace mlir {
+namespace miopen {
+
+struct PipelineConfig {
+  bool highLevelInput = false;
+  bool tuningTest = false;
+  bool runBackend = false;
+  struct PerfConfig {
+    std::string config;
+    int64_t gridSize = 0;
+    int64_t blockSize = 0;
+  } perf;
```

Comment:
I'm not convinced they should belong to pipeline.

For example, if we have two convolutions yet we only have one config/gridSize/blockSize, are we guaranteed that both convolutions can use same of those parameters?

Reply:
I can drop those because they are only used explicitly by mlir-miopen-driver and it only runs mlir::miopen::createAffixTuningParametersPass and does not use the pipeline.

Reply:
That sounds good. Then later we should make them part of the conv2d op.